astropy/coordinates/baseframe.py

```diff
@@ -134,6 +137,214 @@ class RepresentationMapping(NamedTuple):
     defaultunit: str | Unit = "recommended"
 
 
+class CoordinateFrameInfo(MixinInfo):
```

Comment:
This is essentially a copy of the old `SkyCoordInfo`, but with `sc` replaced by `coord` and with a single line removed (which gets added back in the new `SkyCoordInfo`, and with the new `_insert` method.